### PR TITLE
ACMS-1358: Updated Roles in hook update

### PR DESCRIPTION
**Pull request by deepakmishra2**
**Motivation**

Fixes ACMS-1358

**Proposed changes**

**Alternatives considered**

**Testing steps**

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: modules/acquia_cms_headless/acquia_cms_headless.install [hunk 1]
```diff
@@ -33,3 +35,16 @@ function acquia_cms_headless_requirements($phase) {
 
   return $requirements;
 }
+
+/**
+ * Install and update new headless roles.
+ */
+function acquia_cms_headless_update_8001() {
+  \Drupal::configFactory()->getEditable('user.role.headless')->set('label', 'Headless Administrator')->save();
```
Comment (vishalkhode1) [inline modules/acquia_cms_headless/acquia_cms_headless.install:43]: @deepakmishra2 Let's not use Drupal config factory to change label. Use the similar as below i.e
```suggestion
  $role = Role::load('headless');
  if ($role instanceOf Role) {
    //write code to update role label, permissions etc.
  }
```
Same as below update hook as well.

File: modules/acquia_cms_headless/acquia_cms_headless.install [hunk 2]
```diff
@@ -33,3 +35,16 @@ function acquia_cms_headless_requirements($phase) {
...
+  // Create Frontend site with preview role.
+  $role = Role::create([
+    'id' => 'frontend_site_with_preview',
+    'label' => 'Frontend site with preview',
+  ]);
+  $role->save();
```
Comment (vishalkhode1) [inline modules/acquia_cms_headless/acquia_cms_headless.install:49]: @deepakmishra2 This hook is only creating role, but all permissions needs to be added as well. :)

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

File: modules/acquia_cms_headless/acquia_cms_headless.install (hunk 1 above)
Comment (vishalkhode1) [reply modules/acquia_cms_headless/acquia_cms_headless.install:43]: Also, there are 3 more permissions added to headless role. See [this](https://backlog.acquia.com/browse/ACMS-1316) ticket.

File: modules/acquia_cms_headless/acquia_cms_headless.install (hunk 2 above)
Comment (deepakmishra2) [reply modules/acquia_cms_headless/acquia_cms_headless.install:58]: @vishalkhode1 changes are done now, can you review this.

**Comment by deepakmishra2** [comment]
Closing this PR as new pr is raised - https://github.com/acquia/acquia_cms/pull/1225

**Pull request closed without merging**